lightning/src/routing/router.rs

```diff
@@ -2290,7 +2298,7 @@ where L::Target: Logger {
 				// underpaid htlc_minimum_msat with fees.
 				debug_assert_eq!(payment_path.get_value_msat(), value_contribution_msat);
 				value_contribution_msat = cmp::min(value_contribution_msat, final_value_msat);
-				payment_path.update_value_and_recompute_fees(value_contribution_msat);
+				value_contribution_msat += payment_path.update_value_and_recompute_fees(value_contribution_msat);
```

Comment:
FYI, I'm running into an issue with this as part of working on #2570. It seems like the extra fees are counted towards the path's value contribution *and* the path's fees, and that causes us to hit `debug_assert!(*used_liquidity_msat <= hop_max_msat)` below. Still working on a fix.

Reply:
Yeah, it's unpushed so far. Gonna try to get an update out today.

Reply:
Updated. I add 5e742b60cbded6e97dc25dcdea5c695e4d80d521, which differs a bit from this solution^. Let me know if it works for you.